ui/src/importer/importerTaskDetails/ImporterTaskDetailsComponents/ImporterReportStatusLabel.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon, Label } from 'semantic-ui-react';
+
+export class ImporterReportStatusLabel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.labels = {
+      SUCCEEDED: <Label color="green">Import successful</Label>,
+      CANCELLED: <Label color="yellow">Import cancelled</Label>,
+      IMPORTING: (
+        <Label color="blue">
+          <Icon loading name="circle notch" /> Importing literature
+        </Label>
+      ),
+      LOADING: <Label color="grey">Fetching status</Label>,
+      FAILED: <Label color="yellow">Import failed</Label>,
+    };
+  }
+
+  relevantLabel = status => {
+    const statusExists = !!this.labels[status];
+
+    return statusExists ? this.labels[status] : this.labels['FAILED'];
```

Comment:
My only concern is here: We should understand when the status is missing and if it's because of failure, the backend should generate a status. Otherwise, for any potential bug we will end up to the failed label. This is fine from the UI perspective but it just covers the underlying bug! @NRodriguezcuellar could you elaborate when did you notice that the status is empty?

Reply:
That is a fair point! I have not noticed it yet but i think the behaviour before was displaying the failed label if there was no known status so i added that fallback

Reply:
Could you try to replicate the case in which the status is not present? It's worth checking the backend code to see if you can find a case that status is not returned.

Reply:
there are not that many cases actually. Have never seen it not being present

Reply:
Then, I would say to remove it so that the code becomes more explicit! @ntarocco @kprzerwa wdyt?

Reply:
No hard opinion on this one 😄